app/Controllers/Discussions/PostController.php

```diff
@@ -54,7 +59,47 @@ public function create(int $threadId, ?int $postId = null)
             'validator' => $this->validator ?? service('validation'),
         ];
 
-        return $this->render('discussions/posts/create', $data);
+        return view('discussions/posts/_create', $data);
+    }
+
+    /**
+     * Edit post
+     */
+    public function edit(int $postId)
+    {
+        $postModel = model(PostModel::class);
+
+        $post = $postModel->find($postId);
+
+        if (empty($post) ||
+            ($post->author_id !== user_id() &&
+                ! auth()->user()?->can('posts.edit'))) {
+            dd('Unauthorized');
+        }
+
+        if ($this->request->is('put')) {
```

Comment:
I'm not sure how well `put` will work here if we allow uploading images, but we can figure that out when we get there.

Reply:
We will probably do it with a separate request, and then use markdown to embed the file. At least this is how I imagine it now.

Reply:
Makes perfect sense to me.